examples/pytorch/graphsaint/train_sampling.py

```diff
@@ -45,15 +46,25 @@ def main(args):
            n_val_samples,
            n_test_samples))
     # load sampler
+
+    kwargs = {
+        'dn': args.dataset, 'g': g, 'train_nid': train_nid, 'num_workers_sampler': args.num_workers_sampler,
+        'train': False, 'num_subg_norm': args.num_subg_norm, 'batch_size_norm': args.batch_size_norm,
+        'online': args.online, 'num_subg': args.num_subg
+    }
+
     if args.sampler == "node":
-        subg_iter = SAINTNodeSampler(args.node_budget, args.dataset, g,
-                                     train_nid, args.num_repeat)
+        saint_sampler = SAINTNodeSampler(args.node_budget, **kwargs)
     elif args.sampler == "edge":
-        subg_iter = SAINTEdgeSampler(args.edge_budget, args.dataset, g,
-                                     train_nid, args.num_repeat)
+        saint_sampler = SAINTEdgeSampler(args.edge_budget, **kwargs)
     elif args.sampler == "rw":
-        subg_iter = SAINTRandomWalkSampler(args.num_roots, args.length, args.dataset, g,
-                                            train_nid, args.num_repeat)
+        saint_sampler = SAINTRandomWalkSampler(args.num_roots, args.length, **kwargs)
+    else:
+        raise NotImplementedError
+
+    saint_sampler.train = True
```

Comment:
I wonder if we need this flag. Even if we need it for pre-sampling, we can set that internally in the SaintSampler classes.

Reply:
Do you mean we can set this flag as False initially, then when we instantiate the sampler, we run the `__init__` and pre-sample subgraphs, then switch this flag to True?